gnovm/tests/file_test.go

```diff
@@ -70,21 +47,81 @@ func runFileTests(t *testing.T, baseDir string, ignore []string, opts ...RunFile
 
 	opts = append([]RunFileTestOption{WithSyncWanted(*withSync)}, opts...)
 
-	files, err := os.ReadDir(baseDir)
+	files, err := readFiles(t, baseDir)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	files = filterFileTests(t, files, ignore)
-
+	var path string
+	var name string
 	for _, file := range files {
-		file := file
-		t.Run(file.Name(), func(t *testing.T) {
-			runFileTest(t, filepath.Join(baseDir, file.Name()), opts...)
+		path = file.path
+		name = file.DirEntry.Name()
```

Comment:
what about this instead?

```suggestion
		name = strings.TrimPrefix(file.path, baseDir + string(os.PathSeparator))
```

this way we can have a relative path compared to `tests/files/`, which IMO is a better UX when testing (I would expect, reading `TestFiles/declaredType6b_filetest.gno`, that it refers to tests/files/declaredType6b_filetest.gno, but with the current state it actually refers to tests/files/assign_unnamed_type/declaredtype_method/declaredType6b_filetest.gno)

Reply:
additionally, we can remove `type testFile` this way and just make it a `string` (the path)

Reply:
neat, I will leave the `type testFile` for later changes.